renderdoc/driver/vulkan/vk_counters.cpp

```diff
@@ -31,6 +31,75 @@
 #include "driver/ihv/amd/amd_counters.h"
 #include "driver/ihv/amd/official/GPUPerfAPI/Include/GPUPerfAPI-VK.h"
 
+static uint32_t FromKHRCounter(GPUCounter counterID)
+{
+  return (uint32_t) counterID - (uint32_t) GPUCounter::FirstKHR;
+}
+
+static GPUCounter ToKHRCounter(uint32_t idx)
+{
+  return (GPUCounter) ((uint32_t) GPUCounter::FirstKHR + idx);
+}
+
+
+
+void VulkanReplay::convertKhrCounterResult(CounterResult &rdcResult,
+                                           const VkPerformanceCounterResultKHR &khrResult,
+                                           VkPerformanceCounterUnitKHR khrUnit,
+                                           VkPerformanceCounterStorageKHR khrStorage)
+{
+  const auto iterUnit = m_KHRCounterUnitsMap.find(khrUnit);
+  // We should have establish that during the counter enumeration that
+  // everything is already mapped.
+  RDCASSERT(iterUnit != m_KHRCounterUnitsMap.end());
+
+  double value;
+  const KHRCounterMapping &khrMapping = iterUnit->second;
+
+  // Convert everything to doubles.
+  switch (khrStorage)
+  {
+    case VK_PERFORMANCE_COUNTER_STORAGE_INT32_KHR:
+      value = (double) khrResult.int32;
+      break;
+    case VK_PERFORMANCE_COUNTER_STORAGE_UINT32_KHR:
+      value = (double) khrResult.uint32;
+      break;
+    case VK_PERFORMANCE_COUNTER_STORAGE_INT64_KHR:
+      value = (double) khrResult.int64;
+      break;
+    case VK_PERFORMANCE_COUNTER_STORAGE_UINT64_KHR:
+      value = (double) khrResult.uint64;
+      break;
+    case VK_PERFORMANCE_COUNTER_STORAGE_FLOAT32_KHR:
+      value = khrResult.float32;
+      break;
+    case VK_PERFORMANCE_COUNTER_STORAGE_FLOAT64_KHR:
+      value = khrResult.float64;
+      break;
+    default:
+      RDCASSERT(false && "Wrong type");
```

Comment:
Please don't use asserts as a way to print an error message. If there's an unconditional error like this use `RDCERR` to print the error message, and ideally also use the formatting string to print the actual unexpected value (`khrStorage` here).

Reply:
Done.